test/abstract/FlowTransferOperation.sol

```diff
@@ -0,0 +1,452 @@
+            )
+        );
+
+        vm.mockCall(iTokenC, abi.encodeWithSelector(IERC1155.safeTransferFrom.selector), "");
+        vm.expectCall(
+            iTokenC,
+            abi.encodeWithSelector(
+                IERC1155.safeTransferFrom.selector, addressB, addressA, erc1155OutTokenId, erc1155OutAmmount, ""
+            )
+        );
+    }
+
+    function transferERC20ToERC721(address addressA, address addressB, uint256 erc20InAmount, uint256 erc721OutTokenId)
+        internal
+        returns (FlowTransferV1 memory transfer)
+    {
+        transfer = onlyTransferERC20ToERC721(addressA, addressB, erc20InAmount, erc721OutTokenId);
+        mockTransferERC20ToERC721(addressA, addressB, erc20InAmount, erc721OutTokenId);
+    }
+
+    function onlyTransferERC20ToERC721(
+        address addressA,
+        address addressB,
+        uint256 erc20InAmount,
+        uint256 erc721OutTokenId
+    ) internal view returns (FlowTransferV1 memory transfer) {
+        {
+            vm.assume(sentinel != erc20InAmount);
+            vm.assume(sentinel != erc721OutTokenId);
+            assumeAddressNotSentinel(addressA);
+            assumeAddressNotSentinel(addressB);
+        }
+
+        {
+            ERC20Transfer[] memory erc20Transfers = new ERC20Transfer[](1);
+            erc20Transfers[0] =
+                ERC20Transfer({token: address(iTokenA), from: addressA, to: addressB, amount: erc20InAmount});
+
+            ERC721Transfer[] memory erc721Transfers = new ERC721Transfer[](1);
+            erc721Transfers[0] = ERC721Transfer({token: iTokenB, from: addressB, to: addressA, id: erc721OutTokenId});
+
+            transfer = FlowTransferV1(erc20Transfers, erc721Transfers, new ERC1155Transfer[](0));
+        }
+    }
+
+    function mockTransferERC20ToERC721(
+        address addressA,
+        address addressB,
+        uint256 erc20InAmount,
+        uint256 erc721OutTokenId
+    ) internal {
+        vm.mockCall(iTokenA, abi.encodeWithSelector(IERC20.transferFrom.selector), abi.encode(true));
+        vm.expectCall(iTokenA, abi.encodeWithSelector(IERC20.transferFrom.selector, addressA, addressB, erc20InAmount));
+
+        vm.mockCall(iTokenB, abi.encodeWithSelector(bytes4(keccak256("safeTransferFrom(address,address,uint256)"))), "");
+        vm.expectCall(
+            iTokenB,
+            abi.encodeWithSelector(
+                bytes4(keccak256("safeTransferFrom(address,address,uint256)")), addressB, addressA, erc721OutTokenId
+            )
+        );
+    }
+
+    function transferERC721ToERC721(
+        address addressA,
+        address addressB,
+        uint256 erc721InTokenId,
+        uint256 erc721OutTokenId
+    ) internal returns (FlowTransferV1 memory transfer) {
+        transfer = onlyTransferERC721ToERC721(addressA, addressB, erc721InTokenId, erc721OutTokenId);
+        mockTransferERC721ToERC721(addressA, addressB, erc721InTokenId, erc721OutTokenId);
+    }
+
+    function onlyTransferERC721ToERC721(
```

Comment:
the `only*` naming convention is confusing because that's often what modifiers are named i nsolidity